### PR TITLE
Praetorian teamplay buff, making it more useful as a t3

### PR DESCRIPTION
**Pull request by vilereaver41**
## About The Pull Request

Makes Praetorian more effective and deadlier for the hive as a T3 should

## Why It's Good For The Game

Currently, praetorian's neuro spit does absolutely nothing in helping slow a marine push, this is after 12+ matches with the Neuro tweak PR that increases it by 30, I realize the issue is deep enough that the caste itself is just not up to par.

## Changelog
:cl:
balance: neuro spit delay reduced by 0.1 per tier for a total of 1 second at ancient paired with neuro tweak pr
balance: prae's health tweaked ever so slightly
balance: acid spray damage tweaked, now has proper tick damage code placed rather than default.
balance: Acid spray slow down now REMOVED. serves no more purpose, retains 40-second cooldown.
/:cl:

 
after another round of buffs and debates, this prae is now here. Prae is a T3 mini queen and should  at least retain T3 status, the intention with the delay reduction but increased cost is to assist in slowing marine pushes whilst also allowing a retreat ability as marines can run down a prae pre this pr quite easily.

### CODE REVIEW

**Comment by Pariah919** [comment]
>balance: speed increase per tier increased by -0.1+ starting at -0.4 for young
>
no

>balance: plasma increased by 100+ per tier, 1200 at ancient
>
I don't understand why you're increasing the amount of plasma prae has, decreasing praes regen (it has top tier pheros already) won't matter much in the long run as the only thing that uses significant plasma is acid spray which you've only given 200 more cost.
>
>balance: neuro spit delay reduced by 0.1 per tier for a total of 1 second at ancient
>
I don't see the point of this change overall but neurotoxin doesn't do much until you land a lot of it but purge chemicals, and I'd say leaving that niche to sentinel would be better
>balance: prae's health increased slightly per tier
>
nearly as much raw hp than a crusher, better range, better survivablity, nearly queen stats with these changes, big power creep overall, even if it's minor amonts of hp
Also undocumented bullet/melee armor buff for some reason in the middle tiers, If anything prae should have trash tier melee armor, and I don't see the need for midgame praes to be better at their job, t3 xeno already pop out mature, don't need better mid tier.

>balance: acid spray cooldown reduced to 20 seconds from 30 but plasma cost has increased from 200 to 400
>
straight buff, just by this i'd say to never merge this pr, as so far you're basically making the only thing that prevents mass prae hives from being massively viable is their huge CD. This is a massive buff overall to acid spray which really doesn't need to exist, if you're going to buff it halve the slowdown it gives you or 3/4th it.

>balance: acid spray dmg on hit increased, but structure damage decreased.
>
considering you're straight buffing acid spray to be 20 seconds, with a plasma cost you've literally made irrelevant by giving it way more plasma in general. This is a huge no from me bro to both.
>balance: prae plasma regen lowered per tier
>
5-10 points and it gained 100+ plasma as per your words. This isn't a nerf, this is just a straight buff.
>balance: Acid spray slow % and slow timer has been reduced
>
I wouldn't mind just this, honestly this is the only thing prae needs, pit boss mains it which is enough to tell me that it's a decent enough caste.
>add: acid spray tick damage has been added to the ranged attack portion of the caste
>
ok

**Comment by vilereaver41** [comment]
> > balance: speed increase per tier increased by -0.1+ starting at -0.4 for young
> 
> no
> 
> > balance: plasma increased by 100+ per tier, 1200 at ancient
> 
> I don't understand why you're increasing the amount of plasma prae has, decreasing praes regen (it has top tier pheros already) won't matter much in the long run as the only thing that uses significant plasma is acid spray which you've only given 200 more cost.
> 
> > balance: neuro spit delay reduced by 0.1 per tier for a total of 1 second at ancient
> 
> I don't see the point of this change overall but neurotoxin doesn't do much until you land a lot of it but purge chemicals, and I'd say leaving that niche to sentinel would be better
> 
> > balance: prae's health increased slightly per tier
> 
> nearly as much raw hp than a crusher, better range, better survivablity, nearly queen stats with these changes, big power creep overall, even if it's minor amonts of hp
> Also undocumented bullet/melee armor buff for some reason in the middle tiers, If anything prae should have trash tier melee armor, and I don't see the need for midgame praes to be better at their job, t3 xeno already pop out mature, don't need better mid tier.
> 
> > balance: acid spray cooldown reduced to 20 seconds from 30 but plasma cost has increased from 200 to 400
> 
> straight buff, just by this i'd say to never merge this pr, as so far you're basically making the only thing that prevents mass prae hives from being massively viable is their huge CD. This is a massive buff overall to acid spray which really doesn't need to exist, if you're going to buff it halve the slowdown it gives you or 3/4th it.
> 
> > balance: acid spray dmg on hit increased, but structure damage decreased.
> 
> considering you're straight buffing acid spray to be 20 seconds, with a plasma cost you've literally made irrelevant by giving it way more plasma in general. This is a huge no from me bro to both.
> 
> > balance: prae plasma regen lowered per tier
> 
> 5-10 points and it gained 100+ plasma as per your words. This isn't a nerf, this is just a straight buff.
> 
> > balance: Acid spray slow % and slow timer has been reduced
> 
> I wouldn't mind just this, honestly this is the only thing prae needs, pit boss mains it which is enough to tell me that it's a decent enough caste.
> 
> > add: acid spray tick damage has been added to the ranged attack portion of the caste
> 
> ok
> 
> tl;dr this is a straight buff to a caste in a way that will make it masively more oppressive and more annoying by removing it's only limiting factors and the only nerfs i see are literally irrelevant.
> 
> What I propose instead-
> Remove all changes and remove most of the acid spray debuffs that prae has on use, this is a big buff to it as if you use it out of position you go from being instantly dead to slightly bad instead, though prae acid spray has a wide cone of aoe, so honestly I'd say it only needs like 1 slow removed.

your criticism has been taken, and fixes/changes were done

File: code/modules/mob/living/carbon/xenomorph/castes/praetorian/castedatum_praetorian.dm [hunk 1]
```diff
@@ -136,30 +138,31 @@
...
+	melee_damage = 20
 
 	// *** Speed *** //
-	speed = -0.5
+	speed = -0.6
 
 	// *** Plasma *** //
-	plasma_max = 1000
-	plasma_gain = 80
+	plasma_max = 100
```
Comment (InterroLouis) [inline code/modules/mob/living/carbon/xenomorph/castes/praetorian/castedatum_praetorian.dm:147]:
```suggestion
	plasma_max = 950
```

File: code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm [hunk 2]
```diff
@@ -31,10 +31,10 @@
 	X.visible_message(span_xenowarning("\The [X] spews forth a wide cone of acid!"), \
 	span_xenowarning("We spew forth a cone of acid!"), null, 5)
 
-	X.add_movespeed_modifier(type, TRUE, 0, NONE, TRUE, 2)
+	X.add_movespeed_modifier(type, TRUE, 0, NONE, TRUE, 0)
 	start_acid_spray_cone(target, X.xeno_caste.acid_spray_range)
 	add_cooldown()
-	addtimer(CALLBACK(src, .proc/reset_speed), rand(2 SECONDS, 3 SECONDS))
+	addtimer(CALLBACK(src, .proc/reset_speed), rand(0 SECONDS, 0 SECONDS))
```
Comment (KornFlaks) [inline code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm:37]: What is going on here? If you're setting everything to 0, just delete it.

File: code/modules/mob/living/carbon/xenomorph/castes/praetorian/castedatum_praetorian.dm [hunk 3]
```diff
@@ -101,30 +102,31 @@
 	upgrade = XENO_UPGRADE_TWO
 
 	// *** Melee Attacks *** //
-	melee_damage = 23
+	melee_damage = 19
```
Comment (KornFlaks) [inline code/modules/mob/living/carbon/xenomorph/castes/praetorian/castedatum_praetorian.dm:105]: Xenos have 2 damage stages, one at young / mature, another at elder / ancient. Increase this to 20.

File: code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm (hunk 2 above)
Comment (vilereaver41) [reply code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm:37]: the code is there just incase it needs to be used in the future

File: code/modules/mob/living/carbon/xenomorph/castes/praetorian/castedatum_praetorian.dm (hunk 3 above)
Comment (vilereaver41) [reply code/modules/mob/living/carbon/xenomorph/castes/praetorian/castedatum_praetorian.dm:105]: done

File: code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm (hunk 2 above)
Comment (KornFlaks) [reply code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm:37]: Code doesnt work like that. If someone needs to use it in the future, they can add it back in.

Comment (vilereaver41) [reply code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm]: done

File: code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm [hunk 4]
```diff
@@ -31,11 +31,6 @@
 	X.visible_message(span_xenowarning("\The [X] spews forth a wide cone of acid!"), \
 	span_xenowarning("We spew forth a cone of acid!"), null, 5)
 
-	X.add_movespeed_modifier(type, TRUE, 0, NONE, TRUE, 2)
-	start_acid_spray_cone(target, X.xeno_caste.acid_spray_range)
```
Comment (KornFlaks) [inline code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm:35]: Wait, dont actually delete the cone creation method call, and the cooldown call in the next line. The cone and following button cooldown wont start without them.

Also keep your changelog and PR description up to date. You're removing the slowdown following the acid cone usage.

Comment (vilereaver41) [reply code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm:35]: The slowdown is meant to go, this was discussed in the balance disc

Comment (vilereaver41) [reply code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm:35]: Il put the create back in a min

**Comment by Pariah919** [comment]
MJP electric boogaloo

>balance: speed increase per tier increased by -0.1+ starting at -0.4 for young
>
still a no considering prae has queen tier pheromones and long range, as well as decent armor, it doesn't need -0.6 at ancient, if the tradeoff is the health you lose then it's nearly nothing

>balance: plasma per tier now 850 for mature 900 for elder and 1000 for ancient.
>
>balance: neuro spit delay reduced by 0.1 per tier for a total of 1 second at ancient
>
i don't see why prae needs this, give this to sentinel
>balance: prae's health decreased slightly per tier
>
i don't get why you're nerfing praes survablity to justify buffs to acid spray.

>balance: acid spray dmg now stronger on mobs, weaker on structures, now has decent tick dmg and is now 400 plasma.
>
Buffing Praes primary ability to deal even more damage is absurd considering you're removing the slowdown on it which is the primary thing stopping the prae from using it in the open infront of most of marines is kind of absurd, do one or the other.

>balance: prae plasma regen lowered per tier
>
not there so i'm instead going to touch on

>UNDOCUMENTED AS OF COMMENT- Melee damage changes
>
I think these are a terrible way to justify you buffing praes primary ability and just shutter the caste into the reason it's barely played compared to ravager (which nobody can straight face tell you isn't the best t3 post endure) which is that it's kind of boring and unengaging to play. It doesn't need a full stat buff, just minor changes, like movespeed on the acid spray.

>balance: Acid spray slow down now REMOVED. serves no more purpose, for a 30 second ability that can be avoided by running no need to handicap a caste for no good reason
>
you can't avoid acid spray by running anymore, unless you run into someone but if you're using it in such a wide area you kind of don't deserve shit
the acid spray slowdown is there to penalize spamming the ability and you're massively buffing it into powercreep levels, if you want to make prae better give it a slightly better niche as spray is (almost) already good enough as it is imo, the only change i'd make is make the slowdown less, rather than remove it


also the cd is 40 and you made it 30 but the spray lasts for like 5-6 seconds so it feels like 30, so you're reducing it to like 25~

**Comment by David-Stormwell** [comment]
It's a already ok enough caste, don't see why you still want to rebalance it/buff it.
Just a reminder that Rav shouldn't be the new baseline for T3s.

**Comment by SplinterGP** [comment]
Closed on david request for reason above.
image

**Comment by BorranScail** [comment]
@Pariah919
I just want to say that citing a xeno main (pitboss) does not give any credibility to a caste. This player is pro at xeno. This is an outlier. Most players are far below this level is skill and knowledge. Say "skill issue", I dont care. But please do not make comparisons using a single player. 

That is the problem with this reasoning. One player shines as skilled and then xeno reworks are blocked and nerfs are added because "look at this player, he is good. So xenos must be good."
